app/src/main/java/com/andela/gkuti/mipista/tracker/Tracker.java

```
@@ -0,0 +1,191 @@
+    private String location = "unknown";
+    private boolean stop;
+    private BroadcastReceiver locationUpdate;
+
+    /**
+     * Constructor for Tracker class
+     *
+     * @param context
+     */
+    public Tracker(Context context) {
+        this.context = context;
+        datastore = new Datastore(context);
+        userData = new UserData(context);
+        activity = "STI";
+        updateListReceiver = this;
+        locationUpdate = this;
+        activityFilter = new IntentFilter();
+        activityFilter.addAction(Constants.ACTION.getValue());
+        registerLocationUpdates();
+    }
+
+    private void registerLocationUpdates() {
+        locationUpdate = new BroadcastReceiver() {
+            /**
+             * method called when an intent has been received from a broadcast
+             */
+            @Override
+            public void onReceive(Context context, Intent intent) {
+                location = intent.getStringExtra("Location");
+            }
+        };
+        IntentFilter filter = new IntentFilter();
+        filter.addAction("location");
+        context.registerReceiver(locationUpdate, filter);
+    }
+
+    /**
+     * method for starting the tracker
+     */
+    public void startTracker() {
+        stop = false;
+        start();
+        context.registerReceiver(updateListReceiver, activityFilter);
+    }
+
+    /**
+     * method for initializing a new tracking thread
+     */
+    private void start() {
+        if (!isRunning) {
+            initThread();
+            initializeTracking();
+            thread.start();
+        }
+    }
+
+    /**
+     * method that initializes the thread
+     */
+    private void initThread() {
+        thread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                while (activity.equals("STI") && !stop) {
+                    try {
+                        Thread.sleep(1000);
+                    } catch (InterruptedException e) {
```

*Nitpick*: Empty catch block here. Please rectify all instances of this.